### PR TITLE
add AuthAccount model, upd users.json, upd and fix tests

### PR DESCRIPTION
**Pull request by AlexAzalor**

### CODE REVIEW

File: api/routes/auth.py [hunk 1]
```diff
@@ -121,16 +121,16 @@ def apple_auth(auth_data: s.AppleAuthTokenIn, db: Session = Depends(get_db)):
     decoded_token = c.verify_apple_token(auth_data)
 
     user = db.scalar(
-        sa.select(m.User).where(
-            m.User.email == decoded_token.email,
+        sa.select(m.AuthAccount).where(
+            m.AuthAccount.email == decoded_token.email,
```
Comment (Taraslysak) [inline api/routes/auth.py]: you should also filter AuthAcount by type and search only for type apple and oauth_id to match from decoded token

File: api/routes/auth.py [hunk 2]
```diff
@@ -61,13 +61,13 @@ def google_auth(auth_data: s.GoogleAuthIn, db: Session = Depends(get_db)):
         if id_info.iss not in ISSUER_WHITELIST:
             raise ValueError("Wrong issuer.")
 
-        user = db.scalar(sa.select(m.User).where(sa.and_(m.User.email == id_info.email)))
+        user = db.scalar(sa.select(m.AuthAccount).where(sa.and_(m.AuthAccount.email == id_info.email)))
```
Comment (varvarashcherbyna7) [inline api/routes/auth.py]: you also need to add an identifier (from google or apple, or other social networks in the future), together the email, type and identifier must be unique

File: api/routes/auth.py [hunk 3]
```diff
@@ -121,16 +121,16 @@ def apple_auth(auth_data: s.AppleAuthTokenIn, db: Session = Depends(get_db)):
...
+            m.AuthAccount.email == decoded_token.email,
         )
     )
 
     if not user:
         log(log.INFO, "[Google Auth] User [%s] not found. Creating user", decoded_token.email)
         fullname = c.get_apple_fullname(decoded_token)
         user = m.User(
-            email=decoded_token.email,
+            auth_account=[m.AuthAccount(auth_type=s.AuthType.APPLE, email=decoded_token.email)],
```
Comment (Taraslysak) [inline api/routes/auth.py]: you should also add oauth_id to this AuthAccount

File: app/models/auth_account.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    def is_auth_by_google(self):
+        return bool(self.auth_type == AuthType.GOOGLE)
+
+    @property
+    def is_auth_by_apple(self):
+        return bool(self.auth_type == AuthType.APPLE)
+
+    @property
+    def is_auth_by_diia(self):
+        return bool(self.auth_type == AuthType.DIIA)
```
Comment (Taraslysak) [inline app/models/auth_account.py]: I'm not sure we need this

File: app/schema/registration.py [hunk 5]
```diff
@@ -1,10 +1,9 @@
 # Registration data schema
 
-from pydantic import BaseModel, EmailStr
+from pydantic import BaseModel
 
 
 class RegistrationIn(BaseModel):
-    email: EmailStr
     password: str
```
Comment (Taraslysak) [inline app/schema/registration.py]: We still need email when user registers

File: app/schema/user.py [hunk 6]
```diff
@@ -81,6 +79,7 @@ class UsersSearchOut(BaseModel):
 
 
 class UserProfileOut(User):
+    auth_account: list[AuthAccount] = []
```
Comment (Taraslysak) [inline app/schema/user.py]: Info about auth accounts should NEVER leave backend and it should NEVER be sent to a client

File: app/models/user.py [hunk 7]
```diff
@@ -43,14 +44,10 @@ class User(db.Model, ModelMixin):
...
 
-    email: orm.Mapped[str] = orm.mapped_column(sa.String(128))  # fill in registration form
-
     phone: orm.Mapped[str] = orm.mapped_column(sa.String(32), default="")  # fill in registration form
     phone_verified: orm.Mapped[bool] = orm.mapped_column(default=False)
 
-    google_id: orm.Mapped[str] = orm.mapped_column(sa.String(128), default="")
-    apple_id: orm.Mapped[str] = orm.mapped_column(sa.String(128), default="")
-    diia_id: orm.Mapped[str] = orm.mapped_column(sa.String(128), default="")
+    auth_account: orm.Mapped[list["AuthAccount"]] = orm.relationship("AuthAccount", backref="user")
```
Comment (Taraslysak) [inline app/models/user.py]: auth_account => auth_accounts

File: app/schema/user.py [hunk 8]
```diff
@@ -13,14 +15,10 @@ class User(BaseModel):
     uuid: str
     fullname: str
     phone: str = ""
-    email: str
+    auth_account: list[AuthAccount] = []
```
Comment (Taraslysak) [inline app/schema/user.py]: auth_accounts

File: api/controllers/registration.py [hunk 9]
```diff
@@ -16,15 +16,14 @@ def register_user(user_data: s.RegistrationIn, db: Session) -> s.Token:
     if db.scalar(stmt) is not None:
         raise HTTPException(status_code=status.HTTP_406_NOT_ACCEPTABLE, detail="This phone is already registered")
 
-    # check if email is already registered
-    stmt = sa.select(m.User).where(m.User.email == user_data.email)
-    if db.scalar(stmt) is not None:
-        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="This email is already registered")
+    # optional when registering
+    # (hardcoded for now until api router will be created)
+    user_email = ""
```
Comment (Taraslysak) [inline api/controllers/registration.py]: Please remove

File: app/models/auth_account.py (hunk 4 above)
Comment (Taraslysak) [reply app/models/auth_account.py]: Please remove

**Review by Taraslysak — CHANGES_REQUESTED** [review]
Please do some refactoring

File: api/controllers/registration.py [hunk 10]
```diff
@@ -16,14 +16,15 @@ def register_user(user_data: s.RegistrationIn, db: Session) -> s.Token:
     if db.scalar(stmt) is not None:
         raise HTTPException(status_code=status.HTTP_406_NOT_ACCEPTABLE, detail="This phone is already registered")
 
-    # optional when registering
-    # (hardcoded for now until api router will be created)
-    user_email = ""
+    # check if email is already registered
+    stmt = sa.select(m.User).where(m.User.auth_accounts.any(m.AuthAccount.email == user_data.email))
```
Comment (Taraslysak) [inline api/controllers/registration.py]: Check also on AuthAccount type to be basic

**Review by Taraslysak — COMMENTED** [review]
Please consider one more fix

**Pull request merged**